src/tests/multihost/playbook/build_go_ssh_package.yaml

```diff
@@ -0,0 +1,21 @@
+---
+- hosts: localhost
+  vars:
+    repo_loc: "{{ twd }}/../sssd"
+  gather_facts: False
+  tasks:
+    - name: clone easyssh-proxy repo
+      shell: |
+          git clone https://github.com/appleboy/easyssh-proxy.git
+    - name: install easyssh-proxy package
+      shell: |
+          pushd easyssh-proxy # i replaced cd with pushd seems prettier
+          go get -u github.com/appleboy/easyssh-proxy
+          go build -buildmode=c-shared -o /tmp/ssh_library.so {{ repo_loc }}/src/tests/multihost/sssd/testlib/common/remote_ssh.go
+          popd
+    - name: update permission for newly created ssh_library.so
+      shell: |
+          chmod 777 /tmp/ssh_library.so
+      register: build_out
+    - name: print build ouput
+      debug: msg="{{ build_out }}"
```

Comment:
I'd suggest changing from using the shell module to using git for that step and file module for the chmod.  Also you can skip the pushd above by using the chdir args option for the shell module.   Something like this might help:

```
    - name: clone easyssh-proxy repo
      git:
        repo: https://github.com/appleboy/easyssh-proxy.git
        dest: /tmp/easyssh-proxy
        force: yes

    - name: install easyssh-proxy dependencies
      shell: |
        go get -u github.com/appleboy/easyssh-proxy
      args:
        chdir: /tmp/easyssh-proxy

    - name: install easyssh-proxy go module
      shell: |
        go build -buildmode=c-shared -o /tmp/ssh_library.so \
          {{ repo_loc }}/src/tests/multihost/sssd/testlib/common/remote_ssh.go
      args:
        chdir: /tmp/easyssh-proxy
      register: build_out

    - name: print build ouput
      debug: msg="{{ build_out }}"

    - name: update permission for newly created ssh_library.so
      file:
        path: /tmp/ssh_library.so
        mode: '777'
```